File: backend/models/ChitPlan.js
const mongoose = require('mongoose');

const chitPlanSchema = new mongoose.Schema({
  title: {
    type: String,
    required: [true, 'Please provide plan title'],
    trim: true
  },
  amount: {
    type: Number,
    required: [true, 'Please provide chit amount']
  },
  duration: {
    type: Number,
    required: [true, 'Please provide duration in months']
  },
  members_limit: {
    type: Number,
    required: [true, 'Please provide members limit']
  },
  monthly_installment: {
    type: Number,
    required: true
  },
  description: {
    type: String,
    trim: true
  },
  isActive: {
    type: Boolean,
    default: true
  },
  createdBy: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Employee',
    required: true
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('ChitPlan', chitPlanSchema);

File: mobile/lib/models/chit_plan.dart
class ChitPlan {
  final String id;
  final String title;
  final double amount;
  final int duration;
  final int membersLimit;
  final double monthlyInstallment;
  final String? description;
  final bool isActive;

  ChitPlan({
    required this.id,
    required this.title,
    required this.amount,
    required this.duration,
    required this.membersLimit,
    required this.monthlyInstallment,
    this.description,
    required this.isActive,
  });

  factory ChitPlan.fromJson(Map<String, dynamic> json) {
    return ChitPlan(
      id: json['_id'],
      title: json['title'],
      amount: json['amount'].toDouble(),
      duration: json['duration'],
      membersLimit: json['members_limit'],
      monthlyInstallment: json['monthly_installment'].toDouble(),
      description: json['description'],
      isActive: json['isActive'] ?? true,
    );
  }
}
